ts/mains/main_node.ts

```diff
@@ -1023,6 +1024,16 @@ ipc.on('get-start-in-tray', event => {
   }
 });
 
+ipcMain.on('update-badge-count', (_event, count) => {
+  if (app.isReady()) {
+    if (count === 0) {
+      app.setBadgeCount(0); // Clear the badge
+    } else {
+      app.setBadgeCount(count); // Set the badge count
+    }
```

Comment:
ipc calls loose typings, so we are not guaranteed to get a number here.
Even if we did, we could still get a negative number which idk would say.
We could also get a Number.Nan or a Number.Infinity.
```suggestion
      app.setBadgeCount(isNumber(count) && isFinite(count) && count >= 0 ? count : 0); 
```